kratos/linear_solvers/diagonal_preconditioner.h

```diff
@@ -115,31 +115,20 @@ class DiagonalPreconditioner : public Preconditioner<TSparseSpaceType, TDenseSpa
     */
     void Initialize(SparseMatrixType& rA, VectorType& rX, VectorType& rB) override
     {
-        mDiagonal.resize(int(TSparseSpaceType::Size(rX)));
-        mTemp.resize(int(TSparseSpaceType::Size(rX)));
-
-        int i;
+        mDiagonal.resize(static_cast<int>(TSparseSpaceType::Size(rX)));
```

Comment:
```suggestion
        mDiagonal.resize(TSparseSpaceType::Size(rX));
```
no need to overcomplicate ;)